src/Microsoft.DotNet.ScenarioTests.Common/Program.cs

```diff
@@ -249,21 +249,30 @@ private static List<string> CreateBuildTraits(string dotnetRoot, string targetRi
         {
             List<string> buildTraits = new();
 
+            int archSeparatorPos = targetRid.LastIndexOf('-');
+            string ridWithoutArch = targetRid.Substring(0, archSeparatorPos != -1 ? archSeparatorPos : 0);
+            string arch = targetRid.Substring(archSeparatorPos + 1);
```

Comment:
```suggestion
            string[] ridParts = targetRid.Split('-');
            string ridWithoutArch = ridParts[0];
            string arch = ridParts[1];
```

Reply:
This doesn't take arch as the last part (like in `linux-musl-x64`).